### PR TITLE
python 2-->3

### LINKED ISSUE #132: migration from python2 to python3
Hi I just stumbled upon your clarity REST client interface and thought it could be very useful.
However all our code is in python3 
I was wondering if there was any interest in using a python3 version if we make the migration effort

Cheers
Tim

### PR DESCRIPTION
**Pull request by tcezard**
This adds support for python 3 for the core genologics object.
I originally migrated everything but realised that the scripts/example/config were specific to a configuration of clarity so I reverted them back.
I also added a few functions in lims to upload files and move artifacts to a workflow.
I did not get around to add the test I wanted but this is still my todo list.
Hopefully with travis enable we can make sure that any addition is compatible with python2 and python3.

This closes #132  and partially addresses #119

### CODE REVIEW

**Comment by Galithil** [comment]
Quite impressive.

Before I start testing this with our systems, could you revert scripts/tecan_parser.py and bump the version to 0.3.0 ?

**Comment by tcezard** [comment]
I'm not sure why git is seeing differences between the two tecan_parser.py as they're exactly the same as far as I can see.

I'll bump the version up.

**Comment by Galithil** [comment]
vimdiff doesnt catch anything, so it's probably just github being silly.

File: genologics/entities.py [hunk 1]
```diff
@@ -499,8 +503,8 @@ def __setitem__(self, key, value):
                                           nsmap('udf:field'),
                                           type=vtype,
                                           name=key)
-            if not isinstance(value, unicode):
-                value = unicode(str(value), 'UTF-8')
+            if not isinstance(value, str):
+                value = str(str(value), 'UTF-8')
```
Comment (Galithil) [inline genologics/entities.py]: while this is correct in Python 3, str does not take more than one arg in python2.

File: tests/to_rewrite_test_logging.py [hunk 2]
```diff
@@ -1,19 +1,17 @@
...
-from nose.tools import assert_almost_equal, assert_equal, assert_true
-from os.path import isdir,isfile
-from os import listdir
+from os.path import isdir
 import os
 import sys
+from unittest.case import TestCase
 
 from genologics.epp import EppLogger
-import logging
```
Comment (Galithil) [inline tests/to_rewrite_test_logging.py]: Can I get a from __future__ import print_function in here ?

**Comment by tcezard** [comment]
I've fixed the cast as unicode and added tests for this.

**Comment by Galithil** [comment]
I don't see any issue with merging this, the tests I made went fine. 

Is there anything else you want to add to this PR ?

**Comment by tcezard** [comment]
Nope, but there is another one coming

**Comment by Galithil** [comment]
Okay, I'll wait for the next one then.

**Comment by Galithil** [comment]
the pypi package has also been updated. Be sure to open issues should the need arise.

**Comment by tcezard** [comment]
The build is actually failing on python3!
execfile does not exist. (http://stackoverflow.com/questions/436198/what-is-an-alternative-to-execfile-in-python-3-0)
I'll fix this in the next PR

**Comment by Galithil** [comment]
I published a quick fix for python3.

**Comment by Galithil** [comment]
and now I published a fix for python 2 unicode support. https://github.com/SciLifeLab/genologics/pull/156

**Pull request merged**